Modules/_posixsubprocess.c

```diff
@@ -685,9 +685,16 @@ do_fork_exec(char *const exec_array[],
         assert(preexec_fn == Py_None);
 
         pid = vfork();
+        if (pid == -1) {
+            /* If vfork() fails, fall back to using fork(). When it isn't
+             * allowed from a process (such as pid==1 init), vfork can return
+             * -1 with errno EINVAL. https://bugs.python.org/issue47151. */
```

Comment:
The pid 1 usually likes to fork to spawn subprocesses. Is it worth it to remind that vfork() failed to not attempt again (use regular fork() directly)? In this function, fork/vfork is always followed by exec, so if we remind that vfork failed, the child process doesn't inherit this state... since it's a whole process.